fh-messaging/lib/fh_metrics_man.js

```diff
@@ -197,24 +197,66 @@ MetricsManager.prototype.deleteMetricsData = function(topic, date, callback) {
   // if multiple message managers pointed at the same config are running at the
   // same time
   var msgConfig = JSON.parse(JSON.stringify(self.mConfig));
-  var collectionName = topic + "_" + helpers.toYYYYMMDD(date);
-
   messageMan = new fhmsg.Messaging(msgConfig, self.mLogger);
   self.messageMans.push(messageMan);
+
   messageMan.database.on('tearUp', function() {
-    messageMan.database.db.dropCollection(collectionName, callback);
-    //remove all the data that is older than the x number of days from the intermediate tables
-    async.each(['transactionsdesttemp', 'transactionsgeotemp'], function(tempCol, cb) {
-      var daysToCalculate = self.mConfig.agenda.jobs.metrics_rollup_job.daysToKeep || 31;
-      var beforeDate = new Date(date.getFullYear(), date.getMonth(), date.getDate() - daysToCalculate);
-      var query = {"_id.ts":{$lt: beforeDate.getTime()}};
-      messageMan.database.db.collection(tempCol, function(err, collection) {
+    // Pruning for fhact
+    var daysToCalculate = self.mConfig.agenda.jobs.metrics_rollup_job.daysToKeep || 31;
+    var pruning_date = new Date().getTime() - ( daysToCalculate * 24 * 60 * 60 * 1000 );
+    var prune_collection_names = [];
+
+    // Retrieve the list of collections from the database
+    messageMan.database.db.listCollections({name: RegExp("^" + topic + "_")}).toArray( function(err, items) {
```

Comment:
It is missing check if the name exist and is not null

Reply:
If name does not exist or is null then the items returned by the database connector will be empty no?

Reply:
👍 I think it is fine. The find will not return anyway in this conditions.